### PR TITLE
Grouping functionality

### LINKED ISSUE #100: Add grouping functionality
Data is split into train/test sets where a group is only included in the training set or test set.

- [x] Implement code
- [x] Documentation in code (#104 for vignette issue)

### PR DESCRIPTION
**Pull request by zenalapp**
Adds grouping functionality and resolves #100.
Need to describe when this is useful and how to use it in the vignette.

### CODE REVIEW

**Comment by kelly-sovacool** [comment]
Looks like R CMD CHECK threw some warnings, can you fix those @zenalapp?

**Review by kelly-sovacool — APPROVED** [review]
